lib/plugins/aws/package/compile/events/apiGateway/index.js

```diff
@@ -72,9 +72,11 @@ class AwsCompileApigEvents {
 
       // TODO should be removed once AWS fixes the CloudFormation problems using a separate Stage
       'after:deploy:deploy': () => {
+        const getServiceState = require('../../../../lib/getServiceState').getServiceState;
         const updateStage = require('./lib/hack/updateStage').updateStage;
 
-        if (this.validated.events.length === 0) {
+        const state = getServiceState.call(this);
+        if (!this.serverless.utils.isEventUsed(state.service.functions, 'http')) {
```

Comment:
Shouldn't we also rely on resolved `state` in `updateStage` hack?

Reply:
Do you mean rather than using `this.serverless.service` we should use `state.service`?

Reply:
I believe so, otherwise it'll be that for CF deployment we take from `.serverless` folder, and for SDK phase we take from `serverless.yml`.
Both can be out of sync, and I assume we should rely on one config source (?)

Reply:
Yes, you're right. Good catch. Providing a fix in a sec...